tests/cases/Issue110.ts

```diff
@@ -0,0 +1,10 @@
+let number1 = ^^0x100   
+let number2 = ^^14e15
+let number3 = ^^10.42e41.5 
+let number4 = ^^4
+let number5 = ^^51.4
+
```

Comment:
Also try to add the original repro code in #110 to see what if the exponential notation is the second assignment in the same line